upup/pkg/fi/cloudup/azuretasks/vmscaleset.go

```diff
@@ -150,11 +181,23 @@ func (s *VMScaleSet) Find(c *fi.Context) (*VMScaleSet, error) {
 		return nil, fmt.Errorf("failed to parse subnet ID %s", *ipConfig.Subnet.ID)
 	}
 
+	var loadBalancerID *LoadBalancerID
+	for _, i := range *ipConfig.LoadBalancerBackendAddressPools {
+		loadBalancerID, err = ParseLoadBalancerID(*i.ID)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse loadbalancer ID %s", *ipConfig.Subnet.ID)
+		}
+		if !strings.Contains(loadBalancerID.LoadBalancerName, "api") {
+			continue
+		}
```

Comment:
I'm not sure why this is here given that it is at the end of the for loop.  I realize VMSS can have multiple LBs and we're currently only supporting having the one API LB, so maybe this just needs to be moved up above `ParseLoadBalancerID` ?

Reply:
made the change.